packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -145,6 +145,15 @@ function plainFunction(path: NodePath, callId: Object) {
     ]);
     declar._blockHoist = true;
 
+    if (path.parentPath.isExportDefaultDeclaration()) {
+      // change the path type so that replaceWith() does not wrap
+      // the identifier into an expressionStatement
+      path.type = "FunctionExpression";
+      path.parentPath.insertBefore(declar);
+      path.replaceWith(t.identifier(asyncFnId.name));
```

Comment:
`export default x` works as if it creates a secret `default` identifier and does `default = x` when the body is evaluated. It's in TDZ until the body runs, will be an observable read (e.g. call getters), is only assigned exactly once, cannot be referenced within the module's own body.

`export { x as default }` is a regular export of `x` but it's named `default`, which allows the default import sugar and excludes it from being reexported with `export * from`. It's only a declaration, not an assignment, so it does not cause a read of `x`; it will only be in TDZ for as long as `x` itself is in TDZ. If `x` is mutable, reassignments will be visible from outside the module.